File: java/src/test/java/com/kcl/UpdateDependenciesTest.java
package com.kcl;

import com.kcl.api.API;
import com.kcl.api.Spec.ExecProgramArgs;
import com.kcl.api.Spec.ExecProgramResult;
import com.kcl.api.Spec.UpdateDependenciesArgs;
import com.kcl.api.Spec.UpdateDependenciesResult;

import org.junit.Assert;
import org.junit.Test;

public class UpdateDependenciesTest {
    @Test
    public void testUpdateDependencies() throws Exception {
        // API instance
        API api = new API();

        UpdateDependenciesResult result = api.updateDependencies(
                UpdateDependenciesArgs.newBuilder().setManifestPath("./src/test_data/update_dependencies").build());
        Assert.assertEquals(result.getExternalPkgsCount(), 2);
    }

    @Test
    public void testExecProgramWithExternalDependencies() throws Exception {
        // API instance
        API api = new API();

        UpdateDependenciesResult result = api.updateDependencies(
                UpdateDependenciesArgs.newBuilder().setManifestPath("./src/test_data/update_dependencies").build());
        Assert.assertEquals(result.getExternalPkgsCount(), 2);

        ExecProgramArgs execArgs = ExecProgramArgs.newBuilder().addAllExternalPkgs(result.getExternalPkgsList())
                .addKFilenameList("./src/test_data/update_dependencies/main.k").build();

        ExecProgramResult execResult = api.execProgram(execArgs);
        Assert.assertEquals(execResult.getYamlResult(), "a: Hello World!");
    }
}

File: kotlin/src/test/kotlin/com/kcl/UpdateDependenciesTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.execProgramArgs
import com.kcl.api.updateDependenciesArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class UpdateDependenciesTest {
    @Test
    fun testUpdateDependencies() {
        val args = updateDependenciesArgs { manifestPath = "./src/test_data/update_dependencies" }
        val api = API()
        val result = api.updateDependencies(args)
        assertEquals(result.externalPkgsList.size, 2)
    }

    @Test
    fun testExecProgramWithExternalDependencies() {
        val args = updateDependenciesArgs { manifestPath = "./src/test_data/update_dependencies" }
        val api = API()
        val result = api.updateDependencies(args)
        assertEquals(result.externalPkgsList.size, 2)
        val execArgs = execProgramArgs {
            kFilenameList += "./src/test_data/update_dependencies/main.k"
            externalPkgs.addAll(result.externalPkgsList)
        }
        val execResult = api.execProgram(execArgs)
        assertEquals(execResult.yamlResult, "a: Hello World!")
    }
}
